projects/ngrx.io/content/guide/signals/faq.md

```diff
@@ -75,4 +75,74 @@ export class CounterComponent {
   constructor(readonly store: CounterStore) {}
 }
 ```
+
+</details>
+
+<details>
+<summary>I get the error "Cannot assign to read only property 'X' of object '[object Object]'"</summary>
+
+The state in the SignalStore must be immutable. If you make mutable changes, there’s a high risk of introducing subtle, hard-to-diagnose bugs. To protect against this, SignalStore introduced an additional check to enforce immutability.
```

Comment:
Is it worth mentioning which version of SignalStore introduced this? (i.e. v19).

Reply:
Documentation on `ngrx.io` is always up-to-date with the latest version. Docs for previous versions are also available:

- v18: https://v18.ngrx.io/docs
- v17: https://v17.ngrx.io/docs
- etc.